Linking error on some machines due to `zggev3`

## Issue description
Various users have reported compilation errors in the linking step due to `zggev3` not being found.

This routine is the blocked variant of `zggev`, used in the QZ-direct solver. The weird thing is that `zggev` does in fact link and works properly, whereas both routines should be standard in the LAPACK library. Other solvers are also fine.
I'm unable to reproduce this locally

## Bug report
**Minimal example for reproduction**
I'm unable to reproduce this locally, it seems to depend on the machine and/or BLAS/LAPACK library versions though apparently it happens across multiple versions.

**Actual result**
When compiling:
```
Undefined symbols for architecture x86_64:
  "_zggev3_", referenced from:
      ___mod_solvers_MOD_qz_direct in liblego.a(smod_qz_direct.f08.o)
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
```

**Expected result**
Proper compilation and linking.

**Version info**
- Legolas version: 2.0.5

--- a/src/mod_version.f08
+++ b/src/mod_version.f08
@@ -14,6 +14,6 @@ module mod_version
   implicit none
 
   !> legolas version number
-  character(len=10), parameter    :: LEGOLAS_VERSION = "2.0.5"
+  character(len=10), parameter    :: LEGOLAS_VERSION = "2.0.6"
 
 end module mod_version

--- a/src/solvers/smod_qz_direct.f08
+++ b/src/solvers/smod_qz_direct.f08
@@ -61,7 +61,7 @@ contains
     allocate(rwork(8 * N))
     ! get lwork
     allocate(work(1))
-    call zggev3( &
+    call zggev( &
       jobvl, jobvr, N, array_A, lda, array_B, ldb, &
       alpha, beta, vl, ldvl, vr, ldvr, work, -1, rwork, info &
     )
@@ -72,7 +72,7 @@ contains
 
     ! solve eigenvalue problem
     call logger%debug("solving evp using QZ algorithm zggev (LAPACK)")
-    call zggev3( &
+    call zggev( &
       jobvl, jobvr, N, array_A, lda, array_B, ldb, &
       alpha, beta, vl, ldvl, vr, ldvr, work, lwork, rwork, info &
     )

